### ISSUE TITLE
Restructure handling of account group configuration

### ISSUE THREAD

**Issue by hdevalence** [penumbra-zone/web#30]
Currently, the extension creates an IndexedDB instance for view server data records information like the following (using an ephemeral test wallet):
```json
{
    "accounts": {
        "value": [
            {
                "encryptedSeedPhrase": "U2FsdGVkX1/4J7PDteHCf99ZqlYvnOSmC0e9HBeX+rsBSyx86TI2T5EtsUwyTmY+ngsYKBjI1wVcsCcL7+BG+UhvHruKospDwKODt1Z0FVIwU9vasizxhcPVW8CREuFe7U03ek735vpYZ37QDY5FIHoITL6TvokOjHW01IRhjsym5sYzYMFzFYa1NSegtZMom0NO7Q00UyYJDSW9YHC9+UfsSgrfpM88T8bX/b9Hjxc=",
                "label": "Account #1"
            }
        ],
        "version": "V1"
    },
    "hashedPassword": {
        "value": "ade6aaedbe904e48faa3940789930d196b6aa59e981635e22364bb7b2d7ce802",
        "version": "V1"
    }
}
```
This structure has a few different issues, and we should restructure it to better align with the protocol design.

1. **No separate FVK**: We should store the FVK separately, and unencrypted, from the seed phrase. This has two important benefits.  First, it means that the extension can sync block data even when the extension is locked, so that when the user unlocks the extension, it's hopefully already synced to the chain tip. Second, it means that the _viewing capability_ is recorded separately from the _spending capability_, so that all of the viewing-related portions of the code can be completely independent from the spend keys, and in the future, it will be easier to add other custody methods (like a Ledger), since the custody is separate from the scanning infrastructure.
2. **Account terminology mismatch**: The current structure stores one seed phrase (hence spend authority) per "account", but Penumbra provides an integrated logical account mechanism (an "account group" has up to 2^32 numbered accounts per spend key). We want to steer users towards creating multiple logical accounts under the same viewing key, and not to create multiple spending keys, which would duplicate their scanning work.  In addition, if we support multiple spending authorities, we'll need a way to carefully and honestly present that in the UX.

What I'd propose is that we align on the following terminology: use **account** to mean a Penumbra account, and **wallet** to mean a Penumbra account group / spend key.  We will probably want to support multiple "wallets" (e.g., to be able to have a hot wallet with software keys as well as different ledger devices, etc), but we should discourage users from having many wallets by making the extension modal. The extension should have an active wallet, and allow users to pick which wallet is active, but only ever operate with one at a time.  This means that all frontend and UI functionality only needs to present "accounts" to the user, since they will never be operating with two "wallets" at the same time.

With this decision made, we can restructure the configuration data. I'd propose something like:
```json
{
    "wallets": {
        "value": [
            {
                "custody": {
                    "encryptedSeedPhrase": "U2FsdGVkX1/4J7PDteHCf99ZqlYvnOSmC0e9HBeX+rsBSyx86TI2T5EtsUwyTmY+ngsYKBjI1wVcsCcL7+BG+UhvHruKospDwKODt1Z0FVIwU9vasizxhcPVW8CREuFe7U03ek735vpYZ37QDY5FIHoITL6TvokOjHW01IRhjsym5sYzYMFzFYa1NSegtZMom0NO7Q00UyYJDSW9YHC9+UfsSgrfpM88T8bX/b9Hjxc=",
                },
                "fvk": "penumbrafullviewingkey1....",
                "label": "Wallet #1"
            }
        ],
        "version": "V1"
    },
    "activeWalletId": 0,
    "hashedPassword": {
        "value": "ade6aaedbe904e48faa3940789930d196b6aa59e981635e22364bb7b2d7ce802",
        "version": "V1"
    }
}
```
Notes:

* By wrapping the seed phrase inside a `custody` object, we can have other custody methods in the future, or even a "read only" mode where no spending is possible, all using the same code for scanning and syncing;
* Storing the FVK as part of the wallet config means it will always be accessible, independently of the custody method;
* Recording an `activeWalletId` means we know which wallet is the active one;
* All seed phrases are encrypted with the same password, for simplicity.

Currently, the view server data is stored in an IndexedDB instance named `TESTNET-ADDRESS`. Instead, this should be
```
viewdata/TESTNET/NUMERIC_ID
```
e.g.,
```
viewdata/penumbra-testnet-enceladus/0
```
After https://github.com/penumbra-zone/penumbra/issues/2997, we could change the numeric ID to be an account group ID, but this seems potentially overcomplicated for a first pass -- aligning the "account"/"wallet" terminology is more important than the internal database identifiers.

**Comment by grod220** [comment]
Overall, think all of this makes sense. A few comments:

> it means that the extension can sync block data even when the extension is locked, so that when the user unlocks the extension, it's hopefully already synced to the chain tip.

Ideally this would be the case, but I think we rely upon the lifecycle of the service worker to achieve it. I need to dive deeper on this though.

> Recording an activeWalletId means we know which wallet is the active one;

It may be easier to treat the first one in the array to be the active wallet. If not, we could just treat the `fvk` as the unique id. Or perhaps rename `activeWalletId` to `activeWalletIndex`.

**Comment by hdevalence** [comment]
If we want an inherent ID, the two options would be:

- The FVK
- The Account Group ID

The account group ID is a hash of the FVK, so it uniquely identifies the account group but doesn't convey viewing capability. Also it's shorter. The FVK also uniquely identifies its account group, but conveys permanent viewing capability. So the account group ID seems like the better option.

**Comment by grod220** [comment]
Implemented wtih v0.62 updates 👍
